plutus-core/plutus-core/src/PlutusCore/Evaluation/Machine/ExMemory.hs

```diff
@@ -195,12 +195,8 @@ instance ExMemoryUsage Bool where
   memoryUsage _ = 1
 
 -- Memory usage for lists: let's just go for a naive traversal for now.
-instance ExMemoryUsage a => ExMemoryUsage [a] where
-    memoryUsage = sizeList
-        where sizeList =
-                  \case
-                   []   -> 0
-                   x:xs -> memoryUsage x + sizeList xs
```

Comment:
BTW, this is `memoryUsage = foldMap memoryUsage`, right?

Reply:
I think I wanted to be explicit about that so that we didn't get any unexpected surprises.  Didn't we discover that removing annotations (which was going to be happening on the chain) with `<$ ()` was significantly slower than the explicit traversal?  That made me a bit nervous about hiding important things behind library functions.

Reply:
I assumed we didn't care much about performance here, 'cause otherwise it needs to be a strict tail-recursive loop.

Reply:
Good point.  We do care about performance because for instance that function gets called when working out the cost of calls to `equalsData`, which can potentially traverse large structures. I'll fix it and see if it makes any difference to the cost model benchmarks for `equalsData`.

Reply:
I re-ran the costing benchmarks for `equalsData` for this version of `sizeList` and the version you suggested (twice each) and there didn't seem to be any significant difference.  If it did make a big difference then that should show up in the benchmark results because the benchmarks still run the costing function (with an Enormous Budget).  We're running the costing stuff in Haskell and the `equalsData` function in PLC though, so it may be the case that any difference in evaluation time of the size function may be swallowed up in the PLC evaluation time.  So I think we're probably OK, but just to be sure I'll fix the size function the next time I'm doing anything cost-related.

Reply:
Great. Thank you!